### PR TITLE
Rename remaining rules, fix links in guide for built-in rules

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #472

### CODE REVIEW

File: src/Rule/AbstractCompare.php [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+ *
+ * The default comparison is based on string values, which means the values
+ * are compared byte by byte. When comparing numbers, make sure to change {@see AbstractCompare::$type} to
+ * {@see AbstractCompare::TYPE_NUMBER} to enable numeric comparison.
+ *
+ * @see CompareHandler
+ *
+ * @psalm-import-type WhenType from WhenInterface
+ */
+abstract class AbstractCompare implements
```
Comment (samdark) [inline src/Rule/AbstractCompare.php:34]: Makes using constants a bit uglier:

```php
Compare::TYPE_STRING;
AbstractCompare::TYPE_STRING;
```

File: tests/Rule/HasLengthTest.php [hunk 2]
```diff
@@ -6,8 +6,8 @@
 
 use stdClass;
```
Comment (samdark) [inline tests/Rule/HasLengthTest.php:7]: HasLengthTest → LengthTest?

File: src/Rule/AbstractCompare.php (hunk 1 above)
Comment (samdark) [reply src/Rule/AbstractCompare.php:34]: But that's probably OK.

**Review by samdark — APPROVED** [review]

File: src/Rule/AbstractCompare.php (hunk 1 above)
Comment (vjik) [reply src/Rule/AbstractCompare.php:34]: We can create simple class:

```php
final class CompareType {
    const STRING = 'string';
    const NUMBER = 'number';
}
```

This is resolve warning in IDE and make clearly appearance: 

```php
CompareType::STRING
```

**Pull request merged**